File: lab2no1.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int m, n, mStart = 0, nStart = 0;
        String line;

        System.out.println("Enter m:");
        m = scanner.nextInt();
        
        System.out.println("Enter n:");
        n = scanner.nextInt();
        
        scanner.nextLine(); // consume the newline character
        
        System.out.println("Enter line:");
        line = scanner.nextLine();

        int mCurr = 1, nCurr = 1;

        if (m < 1 || n < 1) {
            System.out.println("Error");
        } else {
            for (char i : line.toCharArray()) {
                if (i == 'L') {
                    mCurr -= 1;
                } else if (i == 'R') {
                    mCurr += 1;
                } else if (i == 'D') {
                    nCurr += 1;
                } else if (i == 'U') {
                    nCurr -= 1;
                }
            }

            if (mCurr > m || nCurr > n) {
                System.out.println("Error");
            } else if (mCurr < 1 || nCurr < 1) {
                if (mCurr < 1) {
                    mStart = 2 + Math.abs(mCurr);
                }
                if (nCurr < 1) {
                    nStart = 2 + Math.abs(nCurr);
                }
                System.out.printf("%d %d\n", mStart, nStart);
            } else {
                System.out.println("1 1");
            }
        }
    }
}

File: lab2no1.kt
fun main() {
    var m: Int
    var n: Int
    var mStart = 0
    var nStart = 0
    var isOver = false

    println("Enter m:")
    m = 3
    println("Enter n:")
    n = 3
    println("Enter line:")
    val line = "DDR"

    var mCurr = 1
    var nCurr = 1

    if (m < 1 || n < 1) {
        println("Error")
    } else {
        for (i in line) {
            when (i) {
                'L' -> mCurr -= 1
                'R' -> mCurr += 1
                'D' -> nCurr += 1
                'U' -> nCurr -= 1
            }

            if (mCurr > m || nCurr > n) {
                println("Error")
                isOver = true
                break
            }
        }

        if (mCurr < 1 || nCurr < 1) {
            if (mCurr < 1) {
                mStart = 2 + Math.abs(mCurr)
            }
            if (nCurr < 1) {
                nStart = 2 + Math.abs(nCurr)
            }
            println("$mStart $nStart")
        } else if (!isOver) {
            println("1 1")
        }
    }
}
